src/sagemaker_containers/_files.py

```diff
@@ -132,6 +132,9 @@ def download_and_extract(uri, path):  # type: (str, str) -> None
                 if os.path.exists(path):
                     shutil.rmtree(path)
                 shutil.move(uri, path)
+            elif os.path.exists(uri) and tarfile.is_tarfile(uri):
```

Comment:
You don't need to check if the exists because line 118 creates the path if it does not exist.

Reply:
line 118 checks for the path, not the uri. We should probably move the `os.path.exists(uri)` check before line 135, but after the s3 check on line 122.